projectgenerator/libqgsprojectgen/dataobjects/project.py

```diff
@@ -95,8 +95,7 @@ def create(self, path: str, qgis_project: QgsProject):
 
         qgis_project.setCrs(self.crs)
 
-        if self.legend:
-            self.legend.create(qgis_project)
+        self.legend.create(qgis_project)
```

Comment:
I thought we could make sure that `self.legend` has at least a default value, say a `LegendGroup` corresponding to layer tree root. But since it is better to take the `None` case into account, I've reverted that `if` removal and added another clause to make sure all layers are added directly to layer tree if `self.legend` is `None`.